atomic_reactor/plugins/pre_add_filesystem.py

```diff
@@ -298,12 +312,35 @@ def run(self):
             raise RuntimeError('image task, {}, failed: {}'
                                .format(task_id, task_result))
 
+        return task_id, filesystem_regex
+
+    def stream_filesystem(self, task_id, filesystem_regex):
         filesystem = self.download_filesystem(task_id, filesystem_regex)
 
         new_base_image = self.import_base_image(filesystem)
         self.workflow.builder.set_base_image(new_base_image)
         defer_removal(self.workflow, new_base_image)
 
+        return new_base_image
+
+    def run(self):
+        base_image = self.workflow.builder.base_image
+        if base_image.namespace != 'koji' or base_image.repo != 'image-build':
+            self.log.info('Base image not supported: %s', base_image)
+            return
+
+        image_build_conf = base_image.tag
+        if not image_build_conf or image_build_conf == 'latest':
+            image_build_conf = 'image-build.conf'
+
+        self.session = create_koji_session(self.koji_hub, self.koji_auth_info)
+
+        task_id, filesystem_regex = self.run_image_task(image_build_conf)
+
+        new_base_image = None
+        if not self.is_orchestrator:
+            new_base_image = self.stream_filesystem(task_id, filesystem_regex)
```

Comment:
This would go into test_add_filesystem.py, I'm thinking, where there are some other checks to see that downloadTaskOutput *is* called, but I don't see offhand how to check that it is NOT called. Will have to read more, but pointers appreciated. :)